NEWS.md

```diff
@@ -81,6 +81,8 @@ unit = "s")
 
 14. Added support for `round()` and `trunc()` to extend functionality of `ITime`. `round()` and `trunc()` can be used with argument units: "hours" or "minutes". Thanks to @JensPederM for the suggestion and PR.
 
+15. Support for _cross join_ has been added inside `[.data.table` by providing `on = character()`, closes [#1717](https://github.com/Rdatatable/data.table/issues/1717).
```

Comment:
I would add a few sentences to help distinguish `CJ` from the intended purpose here, something like

> Support for _cross join_ (a.k.a. _Cartesian join_)... That means, in `x[i, on = character()]`, one row is returned for each every row of `x` paired with every row of `i`, a total of `nrow(x) * nrow(i)` rows.